PowerEditor/src/NppCommands.cpp

```diff
@@ -777,8 +789,9 @@ void Notepad_plus::command(int id)
 		break;
 
 		case IDM_VIEW_FUNC_LIST:
+		case IDM_VIEW_SWITCHTO_FUNC_LIST:

```

Comment:
Should IDM_VIEW_SWITCHTO_FUNC_LIST share IDM_VIEW_FUNC_LIST's routine?
I've got a performance here by switching to function list: function list has been launched and displayed, and the focus is on Folder as Workspace panel. After switching to function list by trigger assigned shortcut, function list reloaded and focused. But what I expected is no reload, just focus on it (Like mous clicked).

Reply:
Yes, omitting the reload on re-launches would make the delays go away and the restore tree position would not occur. This would be nicer, for `IDM_VIEW_FUNC_LIST` too. The Function List is rebuilt anyway when the edit buffer switches (at least as long as it is visible). I will look into the mechanisms again and omit the reparse/rebuild process if possible. If the function list changed because of modifications in the edit buffer, then the refresh button can be used to reparse/rebuild.

Reply:
I did three modifications:
- I removed the `reload` call for the Function List from the `DMN_SWITCHIN` notification. 
- Instead, I call `reload` in `launchFunctionList` again, but only, when the Function List is created the first time.
The reload after buffer switching is already done in `Notepad_plus::notifyBufferActivated`.
- I Modified the `cmdID` passed to `launchProjectPanel`: If a Project Panel is started with the new `IDM_VIEW_SWITCHTO_PROJECT_PANEL_n` command, then it gets the corresponding `IDM_VIEW_PROJECT_PANEL_n` as command id. This was necessary for two reasons:
  - Keep the same numbers in config.xml, independent of the command the Project Panel was launched with.
  - The `tTbData.dlgID` contains the toggling commands for all other panels, the Project Panels should do the same.

Reply:
What I meant by "Should IDM_VIEW_SWITCHTO_FUNC_LIST share the same routine of IDM_VIEW_FUNC_LIST?" is that you should do the following without changing the current logic to avoid the regression:
```
		case IDM_VIEW_SWITCHTO_FUNC_LIST:
		{
			if (_pFuncList->isVisible())
			{
				_pFuncList->getFocus();
			}
			else
			{
				checkMenuItem(IDM_VIEW_FUNC_LIST, true);
				_toolBar.setCheck(IDM_VIEW_FUNC_LIST, true);
				launchFunctionList();
				_pFuncList->setClosed(false);
			}
		}
		break;
		
		case IDM_VIEW_FUNC_LIST:
		{
                ...
```

Reply:
Didn't see this while writing my last comment, I will check it out tomorrow.